src/code/system_malloc.c

```diff
@@ -21,53 +23,66 @@ void SystemArena_CheckPointer(void* ptr, u32 size, const char* name, const char*
     }
 }
 
+#define CHECK_POINTER(ptr, size, name, action) SystemArena_CheckPointer(ptr, size, name, action)
+#else
+#define CHECK_POINTER(ptr, size, name, action) (void)0
+#endif
```

Comment:
I know these are defined locally, but I think I would still like if they had a system identifier, like `SYSTEM_ARENA_` and `ZELDA_ARENA_`. It just helps me as Im reading the code to know its doing something locally instead of some generic "check pointer" macro